src/Protector.php

```diff
@@ -501,66 +489,85 @@ protected function getConfigValueForKey(string $key, mixed $default = null): mix
         return is_callable($value) ? $value() : $value;
     }
 
+    /**
+     * Returns the config value for the baseDirectory key.
+     *
+     * @return mixed
+     */
+    public function getBaseDirectory(): mixed
+    {
+        return $this->getConfigValueForKey('baseDirectory');
+    }
+
     /**
      * Prepares the file download response.
      * Prevents the exposure of the connectionName parameter to routing.
      *
      * @param Request $request
      *
-     * @return StreamedResponse
+     * @return Response|StreamedResponse|Application|ResponseFactory
      */
-    public function prepareFileDownloadResponse(Request $request): StreamedResponse
+    public function prepareFileDownloadResponse(Request $request): Response|StreamedResponse|Application|ResponseFactory
     {
         return $this->generateFileDownloadResponse($request);
     }
 
     /**
      * Generates a response which allows downloading the dump file.
      *
-     * @param Request     $request
+     * @param Request $request
      * @param string|null $connectionName
      *
-     * @return StreamedResponse
+     * @return Application|ResponseFactory|Response|StreamedResponse
      */
-    public function generateFileDownloadResponse(Request $request, string $connectionName = null): StreamedResponse
+    public function generateFileDownloadResponse(Request $request, string $connectionName = null): Response|StreamedResponse|Application|ResponseFactory
```

Comment:
do we really want all of these return types

Reply:
Fixed.